enhancements/update/targeted-update-edge-blocking.md

```diff
@@ -0,0 +1,966 @@
+  Additional properties for each entry are defined in [the cluster-condition type registry](#cluster-condition-type-registry).
+
+[The schema version][graph-data-schema-version] would also be bumped to 1.1.0, because this is a backwards-compatible change.
+Consumers who only understand graph-data schema 1.0.0 would ignore the `matchingRules` property and block the edge for all clusters.
+The alternative of failing open is discussed [here](#failing-open).
+
+### Cluster-condition type registry
+
+This registry contains the set of known cluster-condition types, and will be extended if and when future enhancements propose new types.
+
+#### promql
+
+The `type: promql` condition entry is an [object][json-object] with the following additional property:
+
+* `promql` (optional, [string][json-string]), with a [PromQL][] query classifying clusters.
+  This query will be evaluated on the local cluster, so it has access to data beyond the subset that is [uploaded to Telemetry][uploaded-telemetry].
+  The query should return a 1 in the happy case (update should be allowed) and a 0 in the sad case (update should be blocked).
+
+### Enhanced Cincinnati JSON representation
+
+[The Cincinnati graph API][cincinnati-api] will be extended with a new top-level `conditionalEdges` property, with an array of conditional edge [objects][json-object] using the following schema:
+
+* `edges` (required, [array][json-array]), with the update edges covered by this entry.
+  Each entry is an [object][json-object] with the following schema:
+  * `from` (required, [string][json-string]), with the `version` of the starting node.
+  * `to` (required, [string][json-string]), with the `version` of the ending node.
+* `qualifiers` (optional, [array][json-array], with qualifications around the recommendation.
+  Consumers should evaluate all entries, and only recommend the update if there is at least one entry and all entries recommend the update.
+  Each entry is an [object][json-object] with the following schema:
+  * `url` (optional, [string][json-string]), with a URI documenting the issue, as described in [the blocked-edges section](#enhanced-graph-data-schema-for-blocking-edges).
+  * `reason` (optional, [string][json-string]), with a CamelCase reason, as described in [the blocked-edges section](#enhanced-graph-data-schema-for-blocking-edges).
+  * `message` (optional, [string][json-string]), with a human-oriented message describing the blocking reason, as described in [the blocked-edges section](#enhanced-graph-data-schema-for-blocking-edges).
+  * `matchingRules` (optional, [array][json-array]), defining conditions for deciding which clusters have the update recommended and which do not.
+    The array is ordered by decreasing precedence.
+    Consumers should walk the array in order, and stop after the first rule they can successfully evaluate.
+    If no condition can be successfully evaluated, the update should not be recommended.
```

Comment:
same comment as https://github.com/openshift/enhancements/pull/821/files#r705596160

Reply:
GitHub's `/files#r...` links are not stable.  [Here][1]'s a stable `#discussion_r...` link.  And closing this thread in favor of that one, so I can keep track of the outstanding threads, but obviously if we pivot `matchingRules` wording, I'll come through and drop the new wording in all the `matchingRules` laters.

[1]: https://github.com/openshift/enhancements/pull/821#discussion_r705596160